ldms/src/ldmsd/ldmsd_request.c

```diff
@@ -766,6 +767,28 @@ static int string2attr_list(char *str, struct attr_value_list **__av_list,
 	return rc;
 }
 
+int validate_ldmsd_req(ldmsd_req_ctxt_t reqc, ldmsd_req_hdr_t rh)
+{
+	ldmsd_req_attr_t attr;
+	unsigned int off;
+	/* Validate length of request header and request. */
+	if (rh->rec_len <= sizeof(*rh))
+		return 0;
+	attr = (ldmsd_req_attr_t)(rh+1);
+	while (attr->discrim)
+	{
+		off = sizeof(*rh);
```

Comment:
This needs to be moved outside the loop and then updated inside the loop, for example:
```c
off = sizeof(*rh)
while (attr->discrim) {
    off += sizeof(*attr) + attr->attr_len
    if (off > rh->rec_len)
        return 0;
    attr = (ldmsd_req_attr_t)&attr->attr_value[attr->attr_len];
}
```
As it is, it's only checking that the 1st attribute + the length of the longest attribute is not > rh->rec_len.

Reply:
Ahh, ok that makes sense. I'll make the relevant change